proto/desmos/profiles/v1beta1/msgs.proto

```diff
@@ -207,10 +214,42 @@ message BlockUserResponse {}
 
 // MsgUnblockUser represents a message to unblock a previously blocked user.
 message MsgUnblockUser {
-  string blocker = 1 [(gogoproto.moretags) = "yaml:\"blocker\""];
-  string blocked = 2 [(gogoproto.moretags) = "yaml:\"blocked\""];
-  string subspace = 4 [(gogoproto.moretags) = "yaml:\"subspace\""];
+  string blocker = 1 [ (gogoproto.moretags) = "yaml:\"blocker\"" ];
+  string blocked = 2 [ (gogoproto.moretags) = "yaml:\"blocked\"" ];
+  string subspace = 4 [ (gogoproto.moretags) = "yaml:\"subspace\"" ];
 }
 
 // UnblockUserResponse defines the Msg/UnblockUser response type.
 message UnblockUserResponse {}
+
+// ___________________________________________________________________________________________________________________
+
+// MsgLinkAccount represents a message to link an account to a profile.
+message MsgLinkAccount {
```

Comment:
I was thinking again to this message, and I was wondering whether or not we should extend this to also allow users to link their other chain accounts with it. Maybe something like this: 

```proto
message MsgLinkAccount {
  string source_address = 1;
  Proof source_proof = 2;
  ChainConfig source_chain_config = 3;
  string destination_address = 4;
  Proof destination_proof = 5;
}
```

This should allow to connect another chain account (eg. `cosmos1...`) even if the other chain does not yet have implemented the IBC communication with Desmos. The handling will also be identical to the one of the IBC packet. 

What do you think @dadamu @bragaz?

Reply:
I think it is a good idea. Just concern that it seems that we should support different key base type on any other chains ( e.g. `Solana`, `Polkadot`... ) in Desmos cli tool.

Reply:
@RiccardoM I agree with this. @dadamu maybe we can add a field like `key_type` to face this problem....I was already thinking it before reading this comment. Thoughts?

Reply:
I don't get why we should add support for different keys. Aren't those chains using `Secp256k1` keys as well?

Reply:
I'm not sure (@dadamu may know this better) but I remember Solana has a different key type. Correct if I'm wrong

Reply:
As I know, Solana use `ed25519` keys. And it use ".json" key-file while cosmos-based chain use a key base folder to manage all keys.

Reply:
Thanks @dadamu and @bragaz. Knowing this, what I think we can do is handle public keys the same way that Cosmos does in order to handle public keys while signing the data: 

- use `Any` to represent a generic key inside Proto files ([here](https://github.com/cosmos/cosmos-sdk/blob/master/proto/cosmos/tx/signing/v1beta1/signing.proto#L41))
- use the `PubKey` interface to represent a generic key inside Go files ([here](https://github.com/cosmos/cosmos-sdk/blob/master/crypto/types/types.go#L9))

By doing this we can leverage the Cosmos support of all keys types (`secp256k1`, `ed25516` and `secp256r1`) that is included inside the `crypto` package ([here](https://github.com/cosmos/cosmos-sdk/tree/master/proto/cosmos/crypto)) without having to duplicate our code.

Reply:
@RiccardoM It still can't know which signing algorithm is used when receiving a message, I think we can add a filed called algorithm to record it, and remain hex string on pub_key, like this:
message Proof{
  string pub_key = 1;
  string signature = 2;
  string algorithm = 3;
}

Reply:
@dadamu What do you mean with __signing algorithm__? Are other chains using signature algorithms that differ from the Cosmos one (`sign(sha256(message))`)?

Reply:
@RiccardoM like `secp256k1`, `ed25519 ` and `secp256r1`. or it is already included in the proto message?

Reply:
@dadamu There's no need to externally specify the key type. It will be already included inside the JSON encoding of the public key itself as you can see from the following test that I quickly wrote now: 

Proto file definition:
```proto
message Proof {
  google.protobuf.Any public_key = 1 [ (gogoproto.moretags) = "yaml:\"pub_key\"" ];
  string signature = 2 [ (gogoproto.moretags) = "yaml:\"signature\"" ];
}
```

Test code:
```golang
func TestProof_Serialization(t *testing.T) {
  sdk.GetConfig().SetBech32PrefixForAccount("desmos", "desmospub")
  pubKey, err := sdk.GetPubKeyFromBech32(sdk.Bech32PubKeyTypeAccPub, "desmospub1addwnpepq0m3wg0kzfsacgd4tkj8xr622yp9sztwmu22mz0rcu4ep2x5dex65g2v97y")
  require.NoError(t, err)

  any, err := codectypes.NewAnyWithValue(pubKey)
  require.NoError(t, err)

  cdc, _ := app.MakeCodecs()
  json := cdc.MustMarshalJSON(any)
  print(string(json))
}
```

Output:
```
{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"A/cXIfYSYdwhtV2kcw9KUQJYCW7fFK2J48crkKjUbk2q"}
```

Reply:
I've also wrote a small test for future reference to remember how to deserialize that value:

Test code
```golang
func TestKey_Deserialization(t *testing.T) {
  json := `{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"A/cXIfYSYdwhtV2kcw9KUQJYCW7fFK2J48crkKjUbk2q"}`

  cdc, _ := app.MakeCodecs()

  var any codectypes.Any
  cdc.MustUnmarshalJSON([]byte(json), &any)

  var pubkey cryptotypes.PubKey
  err := cdc.UnpackAny(&any, &pubkey)
  require.NoError(t, err)

  print(pubkey.String())
}
```

Output:
```
PubKeySecp256k1{03F71721F61261DC21B55DA4730F4A510258096EDF14AD89E3C72B90A8D46E4DAA}
```

Reply:
@RiccardoM I test it with ed25519 key. And it returns error, it shows that the pubkey of ed25519 doesn't implement proto.Message

Code:
```go
func main() {
	priv := ed25519.GenPrivKey()
	pubKey := priv.PubKey()

	any, _ := codectypes.NewAnyWithValue(pubKey)

	cdc, _ := simapp.MakeCodecs()
	json := cdc.MustMarshalJSON(any)
	fmt.Println(string(json))
}
```

Error:
```
cannot use pubKey (type "github.com/tendermint/tendermint/crypto".PubKey) as type "github.com/gogo/protobuf/proto".Message in argument to "github.com/cosmos/cosmos-sdk/codec/types".NewAnyWithValue:
        "github.com/tendermint/tendermint/crypto".PubKey does not implement "github.com/gogo/protobuf/proto".Message (missing ProtoMessage method)
```

Reply:
I wrote small test for `ed25519`, it works, too.

Code:
```go
func main() {
	priv := ed25519.GenPrivKey()
	pubKey := priv.PubKey()

	any, _ := codectypes.NewAnyWithValue(pubKey)

	cdc, _ := simapp.MakeCodecs()
	json := cdc.MustMarshalJSON(any)
	fmt.Println(string(json))
}
```
Output:
```
{"@type":"/cosmos.crypto.ed25519.PubKey","key":"S1s1TvGA+RLDJViBSO7kX1lhTXAbV3h7rMBvb8FI7DE="}
```

Reply:
@RiccardoM So, other non cosmos-based chains can also import the packet and create a proof whose public key includes type like `{"@type":"/cosmos.crypto.ed25519.PubKey","key":"S1s1TvGA+RLDJViBSO7kX1lhTXAbV3h7rMBvb8FI7DE="}`, then it can be deserialized the public key to verify the signature on Desmos, right?

Reply:
@dadamu That's correct